### PR TITLE
Feature/restructure code

### PR DESCRIPTION
**Pull request by sven1103**

### CODE REVIEW

**Review by luiskuhn — APPROVED** [review]
looks good

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,19 @@
 # Changelog
```
Comment (wow-such-code) [inline CHANGELOG.md:1]: should probably be updated relating omero

File: drop-boxes/register-omero-metadata/backendinterface.py [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+imaging data (raw and metadata) form the OMERO server (v5.4).
+It requires that the following software be installed within the Python
+environment you are loading this module:
+	* OMERO.cli (https://docs.openmicroscopy.org/omero/5.4.0/users/cli/index.html)
+	* OMERO Python language bindings (https://docs.openmicroscopy.org/omero/5.4.0/developers/Python.html)
+This code is based on the following documentation:
+    https://docs.openmicroscopy.org/omero/5.4.0/developers/Python.html
+It contains the following functions:
+    * omero_connect - connects to server
+    * TODO...
```
Comment (wow-such-code) [inline drop-boxes/register-omero-metadata/backendinterface.py:13]: are there more functions to describe?

File: drop-boxes/register-omero-metadata/backendinterface.py [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+        print("std_out: " + std_out)
+        print("std_err: " + std_err)
+        print("return_code: " + str(proc.returncode))
+
+        if int(proc.returncode) == 0:
+
+            print("-->" + std_out)
+
+            fist_line = std_out.splitlines()[0]
+            image_ids = fist_line[6:].split(',')
```
Comment (wow-such-code) [inline drop-boxes/register-omero-metadata/backendinterface.py:147]: this looks dangerous

Comment (wow-such-code) [reply drop-boxes/register-omero-metadata/backendinterface.py:147]: also "first"*
Would it be possible to do this via regex?

File: drop-boxes/register-omero-metadata/backendinterface.py [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            shell=True,
+                            universal_newlines=True)
+
+        std_out, std_err = proc.communicate()
+
+        if int(proc.returncode) == 0:
+
+            fist_line = std_out.splitlines()[0]
```
Comment (wow-such-code) [inline drop-boxes/register-omero-metadata/backendinterface.py:198]: same as above, also maybe put it in its own method

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        process = Popen( "/bin/bash", shell=False, universal_newlines=True, stdin=PIPE, stdout=PIPE, stderr=PIPE )
+        out, err = process.communicate( commands )
+
+        id_list = str(out).split()
+
+        return id_list
+        
+
+    def triggerOMETiffConversion(self):
+        pass
```
Comment (wow-such-code) [inline drop-boxes/register-omero-metadata/image_registration_process.py:92]: what does this do?

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+                    tsvFileMap.update(row)
+        except IOError:
+            print "Error: No file found at provided filepath " + tsvFilePath
+        except csv.Error as e:
+            print 'Could not gather the Metadata from TSVfile %s, in line %d: %s' % (tsvfile, reader.line_num, e)
+
+        return tsvFileMap
+
+    def registerExperimentDataInOpenBIS(self):
+        pass
```
Comment (wow-such-code) [inline drop-boxes/register-omero-metadata/image_registration_process.py:110]: if the function is not needed it should be removed

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 7]
```diff
@@ -122,70 +107,120 @@ def callOmeroWithFilePath(file_path, sample_barcode):
...
+	# 3. We now request the associated omero dataset id for the openBIS sample code.
+	# Each dataset in OMERO contains the associated openBIS biological sample id, which
+	# happened during the experimental design registration with the projectwizard.
+	omero_dataset_id = registrationProcess.requestOmeroDatasetId()
+
+	# Find and parse metadata file content
+	metadataFile = findMetaDataFile(incomingPath)
+	
+	# Iterate over the metadata entries containing all pre-specified imaging metadata
+	for line in metadataFile[1:]:  # (Exclude header)
```
Comment (wow-such-code) [inline drop-boxes/register-omero-metadata/register-omero.py:179]: some metadata examples should be presented imho

File: drop-boxes/register-omero-metadata/backendinterface.py (hunk 3 above)
Comment (sven1103) [reply drop-boxes/register-omero-metadata/backendinterface.py:147]: indeed. but this was the code that has been used and we reference this for the publication. 

Regex would be a safer way to go here. Would you mind and open an issue @wow-such-code such that we can keep track of it?

**Pull request merged**